sdk/tutorials/create-wallet-ai-agent.md

```diff
@@ -0,0 +1,299 @@
+
+## Prerequisites
+
+- [Node.js](https://nodejs.org/) version 18 or later
+- An [OpenAI](https://platform.openai.com/docs/overview) API key
+- [MetaMask](https://metamask.io/) installed, with an account that has Linea Sepolia ETH
+  :::note
+  You can use the [MetaMask faucet](/developer-tools/faucet) to get Linea Sepolia ETH.
+  :::
+
+## Steps
+
+### 1. Set up the project
+
+Clone the [`Consensys/wallet-agent`](https://github.com/Consensys/wallet-agent/tree/main) repository, switch to the `initial-setup` branch, and install dependencies:
+
+```bash
+git clone git@github.com:Consensys/wallet-agent.git
+cd wallet-agent
+git switch initial-setup
+npm install
+```
+
+In the root of your project, create a `.env.local` file:
+
+```bash
+touch .env.local
+```
+
+In `.env.local`, add an `OPENAI_API_KEY` environment variable, replacing `<YOUR-API-KEY>` with your [OpenAI](https://platform.openai.com/docs/overview) API key.
+Vercel's AI SDK will use this environment variable to authenticate your dapp with the OpenAI service.
+
+```text title=".env.local"
+OPENAI_API_KEY=<YOUR-API-KEY>
+```
+
+### 2. Create the dapp interface
+
+In `app/page.tsx`, use the `useAccount`, `useConnect`, and `useDisconnect` hooks from Wagmi, along with the Wagmi [MetaMask SDK connector](https://wagmi.sh/react/api/connectors/metaMask) to create a button to connect and disconnect your MetaMask wallet.
+
+Use the `Chat` component to display the AI agent chat interface.
+
+```tsx title="page.tsx"
+// add-next-line
++ "use client";
+
+// add-start
++ import { useAccount, useConnect, useDisconnect } from "wagmi";
++ import { metaMask } from "wagmi/connectors";
++ import { Button } from "@/components/ui/button";
++ import { Chat } from "@/components/Chat";
+// add-end
+import Image from "next/image";
+
+// add-start
++ const ConnectButton = () => {
++   const { connect } = useConnect();
++   const { address, isConnected } = useAccount();
++   const { disconnect } = useDisconnect();
++ 
++   return (
++     <div className="mx-auto">
++       {isConnected ? (
++         <Button onClick={() => disconnect()}>Disconnect {address}</Button>
++       ) : (
++         <Button onClick={() => connect({ connector: metaMask() })}>Connect</Button>
++       )}
++     </div>
++   );
++ };
+// add-end
+
+export default function Home() {
+// add-next-line
++ const { isConnected } = useAccount();
+  return (
+    <div className="h-screen w-full overflow-y-auto grid grid-rows-[20px_1fr_20px] items-center justify-items-center min-h-screen p-8 pb-20 gap-16 sm:p-20 font-[family-name:var(--font-geist-sans)]">
+      <main className="gap-8 row-start-2 sm:items-start h-full w-full">
+        <h1>Wallet Agent setup</h1>
+// add-start
++       <ConnectButton />
++       { isConnected ? <Chat /> : null}
+// add-end
+      </main>
+      // ...
+```
+
+To test the interface, run the development server and navigate to [`http://localhost:3000`](http://localhost:3000/):
+
+```bash
+npm run dev
```

Comment:
FYI, this initially failed because I didn't have `pnpm` installed. 

Once, I installed `pnpm`, I then received an error because native SWC binaries for my platform (@next/swc-darwin-arm64) were not installed -- probably not worth mentioning though.